### PR TITLE
Charmis/plugin upgradation

### PR DESCRIPTION
**Pull request by CharmisG**
- Plugin upgradation 
- Converted code from java to kotlin for child (example) folder

### CODE REVIEW

File: android/src/main/java/sqip/flutter/SquareInAppPaymentsFlutterPlugin.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- Copyright 2018 Square Inc.
+ Copyright 2022 Square Inc.
```
Comment (hukid) [inline android/src/main/java/sqip/flutter/SquareInAppPaymentsFlutterPlugin.java:2]: nit, we don't have to change this header.

File: android/build.gradle [hunk 2]
```diff
@@ -25,8 +25,8 @@ rootProject.allprojects {
 apply plugin: 'com.android.library'
 
 def MIN_IAP_SDK_VERSION = '1.5.6'
-def COMPILE_SDK_VERSION = 28
-def TARGET_SDK_VERSION = 28
+def COMPILE_SDK_VERSION = 30
```
Comment (hukid) [inline android/build.gradle:28]: Do we have to bump up the target SDK version? Is that required by Flutter?

File: example/android/build.gradle [hunk 3]
```diff
@@ -8,7 +8,7 @@ buildscript {
 
     dependencies {
         classpath 'com.android.tools.build:gradle:4.2.2'
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.7.10"
```
Comment (hukid) [inline example/android/build.gradle:11]: Why do we need to change this?

**Review by hukid — APPROVED** [review]

File: android/build.gradle (hunk 2 above)
Comment (CharmisG) [reply android/build.gradle:28]: No but we are upgrading for all so i have upgrade this also.

File: android/src/main/java/sqip/flutter/SquareInAppPaymentsFlutterPlugin.java (hunk 1 above)
Comment (CharmisG) [reply android/src/main/java/sqip/flutter/SquareInAppPaymentsFlutterPlugin.java:2]: Okay I have undo this.

**Pull request merged**